### PR TITLE
feat: Add edit and download themes from github repo

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[colors--canary.276.4013191131.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.276.4013191131.xml)  
[color_metro_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.276.4013191131.swift)  
[color_metro_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.276.4013191131.kt)  
[color_metro_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.276.4013191131.ts)  
[color_sbermarket_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.276.4013191131.swift)  
[color_sbermarket_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.276.4013191131.kt)  
[color_sbermarket_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.276.4013191131.ts)  
[color_sberprime_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.276.4013191131.swift)  
[color_sberprime_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.276.4013191131.kt)  
[color_sberprime_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.276.4013191131.ts)  
[color_selgros_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.276.4013191131.swift)  
[color_selgros_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.276.4013191131.kt)  
[color_selgros_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.276.4013191131.ts)  
[color_smbusiness_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.276.4013191131.swift)  
[color_smbusiness_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.276.4013191131.kt)  
[color_smbusiness_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.276.4013191131.ts)  
[PlasmaTokensColor--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.276.4013191131.swift)  
[typo_mage_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.276.4013191131.swift)  
[typo_mage_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.276.4013191131.kt)  
[typo_mage_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.276.4013191131.ts)  
[typo_plasma_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.276.4013191131.swift)  
[typo_plasma_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.276.4013191131.kt)  
[typo_plasma_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.276.4013191131.ts)  
[typo_ruler_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.276.4013191131.swift)  
[typo_ruler_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.276.4013191131.kt)  
[typo_ruler_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.276.4013191131.ts)  
[typo_sage_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.276.4013191131.swift)  
[typo_sage_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.276.4013191131.kt)  
[typo_sage_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.276.4013191131.ts)  
[typo_sbermarket_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.276.4013191131.swift)  
[typo_sbermarket_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.276.4013191131.kt)  
[typo_sbermarket_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.276.4013191131.ts)  
[typo_soulmate_ios-swift--canary.276.4013191131.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.276.4013191131.swift)  
[typo_soulmate_kotlin--canary.276.4013191131.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.276.4013191131.kt)  
[typo_soulmate_react-native--canary.276.4013191131.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.276.4013191131.ts)

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-276/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-276/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-276/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-276/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-276/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-276/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-276/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-276/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-276/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-276/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-276/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-276/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-276/temple-storybook/

File: website/plasma-theme-builder/src/components/Token/Token.tsx [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
+import { getHEXA, getRGBA, TokenContext } from '../utils';
+import { IconTrashFilled, IconEye, IconEdit } from '@salutejs/plasma-icons';
+import { iconButtonFade } from '../mixins';
 import type { TokenValue } from '../types';
 
+const IconButtons = styled.div`
+    display: flex;
+`;
+
+const IconButton = styled.div`
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/Token/Token.tsx:14]: почему это это не Button с Icon ?

File: website/plasma-theme-builder/src/components/Token/Token.tsx [hunk 2]
```diff
@@ -62,31 +75,102 @@ const hasOrigin = (values: string | Record<string, any>): values is Record<strin
...
-
+export const Token = ({ section, subsection, name, data }: TokenProps) => {
+    const { onOpenTokenForm, onTokenDelete, onTokenEnabled } = useContext(TokenContext);
     const [visible, setVisible] = useState(false);
 
-    const newValue = (!hasOrigin(value) && isGradient(value)) || hasOrigin(value) ? '#FFFFFFFF' : value;
+    const { value, comment = '', enabled } = data;
+
+    const normalizeValue = useMemo(
+        () => ((!hasOrigin(value) && isGradient(value)) || hasOrigin(value) ? '#FFFFFFFF' : value),
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/Token/Token.tsx:91]: мой мозг сломался на этом однострочнике

File: website/plasma-theme-builder/src/components/Token/Token.tsx [hunk 3]
```diff
@@ -62,31 +75,102 @@ const hasOrigin = (values: string | Record<string, any>): values is Record<strin
...
+
+    const onTokenDeleteClick = useCallback(() => {
+        onTokenDelete(tokenInfo);
+    }, [onTokenDelete, tokenInfo]);
+
+    const onTokenEnabledClick = useCallback(() => {
+        onTokenEnabled({
+            ...tokenInfo,
+            enabled: {
+                value: enabled === undefined || enabled === true ? false : true,
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/Token/Token.tsx:137]: тут тоже сломался

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    display: flex;
+    gap: 1rem;
+`;
+
+const PreviewColor = styled.div<{ value: string }>`
+    width: 3rem;
+    height: 3rem;
+    border-radius: 0.75rem;
+
+    ${({ value }) =>
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:25]: а что будет если я напишу:
```tsx
styled.div`
    width: 3rem;
    height: 3rem;
    border-radius: 0.75rem;
    background: ${({ value }) => value }
`
```

Comment (Yeti-or) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:25]: и в целом value же наверно background-color а не весь background ?

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                name: {
+                    value: inputData.name.value,
+                    status: 'error',
+                    helpText: 'Название токена должно начинаться с ' + section.value,
+                },
+            });
+            return;
+        }
+
+        if (!RegExp(/#\b[0-9A-Fa-f]{8}\b/g).test(value.value)) {
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:123]: всё ещё не понимаю за конструктор

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+            onInputDataChange({
+                ...inputData,
+                value: {
+                    value: inputData.value.value,
+                    status: 'error',
+                    helpText: 'Цвет должен быть в формате #FFFFFFF',
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:129]: у тебя тут 7 символов `F` а регулярка требует ровно 8

File: website/plasma-theme-builder/src/components/mixins.ts [hunk 7]
```diff
@@ -3,3 +3,13 @@ import { css } from 'styled-components';
 export const SBSansTextMono = css`
     font-family: 'SB Sans Text Mono';
 `;
+
+export const iconButtonFade = css`
+    cursor: pointer;
+
+    &&&:hover {
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/mixins.ts:10]: ???

File: website/plasma-theme-builder/src/components/Token/Token.tsx (hunk 1 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Token/Token.tsx:14]: хорошее предложение, спасиб :ok

File: website/plasma-theme-builder/src/components/Token/Token.tsx (hunk 2 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Token/Token.tsx:91]: разделил, надеюсь более читаемо стало

File: website/plasma-theme-builder/src/components/Token/Token.tsx (hunk 3 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Token/Token.tsx:137]: тут што-то хз, как упросить понимание. Смысл в том, что нужна проверка, если enabled принимает значение или undefined или true, тогда токен считается включенным. Первая проверка на undefined нужна для тем, которые уже созданы у нас (sbermarket, sberprime) и которые не имеют поле enabled

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx (hunk 4 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:25]: так тоже можно, спасиб 👌

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx (hunk 5 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:123]: по сути это regex для HEX значений

Comment (Yeti-or) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:126]: я про вот это:

```
/#\b[0-9A-Fa-f]{8}\b/g.test('#c0ffeeaf')
true
RegExp(/#\b[0-9A-Fa-f]{8}\b/g).test('#c0ffeeaf')
true
```

File: website/plasma-theme-builder/src/components/mixins.ts (hunk 7 above)
Comment (Yeti-or) [reply website/plasma-theme-builder/src/components/mixins.ts:10]: https://codesandbox.io/s/plasma-web-example-opacity-yesffk?file=/src/App.js

так не годится ?

**Review by Yeti-or — APPROVED** [review]

File: website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx (hunk 5 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/TokenForm/TokenForm.tsx:126]: спасибо, действительно тут можно без него

File: website/plasma-theme-builder/src/components/mixins.ts (hunk 7 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/mixins.ts:10]: вообще заход хороший, но основная проблема в том, что я подсвечиваю именно иконки, а не кнопки, и из-за этого нужно так кастыльно переписывать специфичность. Но в принципе, я могу всё это и на кнопочках попробовать сделать

File: website/plasma-theme-builder/src/components/Token/Token.tsx (hunk 3 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Token/Token.tsx:146]: их prettier сносит :C

Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Token/Token.tsx:146]: немного упростил, отказавшись от тернарника => 

```ts
                value: !(enabled === undefined || enabled === true),
```

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-276/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🔴 Fail**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-276/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-276/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-276/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-276/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-276/temple-storybook/

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-276/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-276/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-276/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-276/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-276/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-276/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-276/temple-storybook/

**Pull request merged**